mitodl/open-discussions
# add transcripts

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/2375

#### What's this PR do?
This pr adds functions to download and parse the transcript for a video.

#### How should this be manually tested?
```
  from course_catalog.models import *
  from course_catalog.etl import youtube
  v = Video.objects.all()[0]
  c = youtube.get_captions_for_video(v)
```
Should generate the captions in xml format

```
youtube.parse_video_captions(c)
```

Should parse the transcript from the captions
#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?

Generating transcripts is fairly slow. On my machine, I was able to process the transcripts for only 30 videos per minute. Transcripts will need to be generated and updated in a separate cron job from the rest of the youtube video etl

#### Screenshots (if appropriate)
(Optional)

#### What GIF best describes this PR or how it makes you feel?
(Optional)
